cabal-helper.cabal

```diff
@@ -186,7 +183,6 @@ test-suite ghc-session
   build-depends:       ghc              < 8.9  && >= 8.0.2
                      , pretty-show      < 1.9  && >= 1.8.1
                      , cabal-helper
```

Comment:
this works with `cabal build --enable-tests`? i was giving a try and it failed for me in `ghc-session` with:
```
tests\GhcSession.hs:71:31: error:
    • Couldn't match expected type ‘CabalHelper.Compiletime.Types.Programs’
                  with actual type ‘Programs’
      NB: ‘Programs’
            is defined in ‘CabalHelper.Compiletime.Types’
                in package ‘cabal-helper-1.0.0.0’
          ‘CabalHelper.Compiletime.Types.Programs’
            is defined in ‘CabalHelper.Compiletime.Types’
    • In the first argument of ‘modProgs’, namely ‘defaultPrograms’
      In the expression: modProgs defaultPrograms
      In the expression:
        let
          ?verbose = const False
          ?progs = modProgs defaultPrograms
        in action
   |
71 |             ?progs = modProgs defaultPrograms
   |  
```
So i was about to remove `cabal-helper`from `build-depends` and add the public lib modules as src ones

Reply:
Didnt test, sorry, will do immediately

Reply:
This seems non-trivial as this depends on both the internal common stanza and cabal-helper. they need to depend on the same common-stanza, which cabal doesnt seem to be able to prove (which sounds like the exact reason for private libraries).

Reply:
> So i was about to remove cabal-helper from build-depends and add the public lib modules as src ones
Have you tried this?

Reply:
Yeah looks you'll have to the main cabal-helper library into a stanza too :/